applications/asset_tracker/src/main.c

```diff
@@ -1856,5 +1903,8 @@ void main(void)
 	k_sem_take(&cloud_ready_to_connect, K_FOREVER);
 #endif
 
+	date_time_update_async(date_time_event_handler);
+	k_sem_take(&date_time_obtained, K_FOREVER);
```

Comment:
Let's have a timeout and check the return code of `k_sem_take()`

Reply:
Agreed, added timeout, and check for ```aws_iot``` sample and ```asset_tracker```.